src/filesystem.cpp

```diff
@@ -106,6 +106,19 @@ bool FS::rename(const Path &from, const Path &to)
   return !func(nativePath(from).c_str(), nativePath(to).c_str());
 }
 
+bool FS::symlink(const std::string &target, const Path &link)
+{
+  mkdir(link.dirname());
+
+  const auto &nativeLink = nativePath(link);
+
+#ifdef _WIN32
+  return CreateSymbolicLink(nativeLink.c_str(), target.c_str(), 0x0);
```

Comment:
I'm not sure what to do here... I've tried a bunch of different variations. I don't have a windows machine, so it takes ~15 minutes for CI to provide feedback.

Reply:
The Windows API needs UTF-16 wchar_t strings instead of UTF-8 char strings:
```cpp
return CreateSymbolicLink(nativeLink.c_str(), Win32::widen(target).c_str(), 0);
```
Also I'm pretty sure Windows requires the process to gain admin privileges to be allowed to create symlinks... (And the `SYMBOLIC_LINK_FLAG_ALLOW_UNPRIVILEGED_CREATE` flag requires Developer Mode to be enabled.) If so I don't think this will be acceptable as a feature, or it will need to only be available on Linux/macOS...

Anyway, I'll check & review the PR more closely later on. Thanks for the PR!

Reply:
Thanks! I changed it to use `Path` and now the Windows built went green.

I hope you will consider including it, even if we have to disable support for it on Windows. It's super handy.